src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -69,27 +69,42 @@ def validate_location(cmd, location):
             break
 
 
-def get_chart_path(registry_path, kube_config, kube_context, helm_client_location):
+def get_chart_path(registry_path, kube_config, kube_context, helm_client_location, for_preonboarding_checks=False):
     # Pulling helm chart from registry
     os.environ['HELM_EXPERIMENTAL_OCI'] = '1'
-    pull_helm_chart(registry_path, kube_config, kube_context, helm_client_location)
+    pull_helm_chart(registry_path, kube_config, kube_context, helm_client_location, for_preonboarding_checks)
 
     # Exporting helm chart after cleanup
-    chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'AzureArcCharts')
+    if for_preonboarding_checks:
+        chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'ConnectPrecheckCharts')
+    else:
+        chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'AzureArcCharts')
     try:
         if os.path.isdir(chart_export_path):
             shutil.rmtree(chart_export_path)
     except:
-        logger.warning("Unable to cleanup the azure-arc helm charts already present on the machine. In case of failure, please cleanup the directory '%s' and try again.", chart_export_path)
-    export_helm_chart(registry_path, chart_export_path, kube_config, kube_context, helm_client_location)
+        if for_preonboarding_checks:
+            logger.warning("Unable to cleanup the connect-precheck helm charts already present on the machine. In case of failure, please cleanup the directory '%s' and try again.", chart_export_path)
+        else:
+            logger.warning("Unable to cleanup the azure-arc helm charts already present on the machine. In case of failure, please cleanup the directory '%s' and try again.", chart_export_path)
+    export_helm_chart(registry_path, chart_export_path, kube_config, kube_context, helm_client_location, for_preonboarding_checks)
 
     # Returning helm chart path
-    helm_chart_path = os.path.join(chart_export_path, 'azure-arc-k8sagents')
-    chart_path = os.getenv('HELMCHART') if os.getenv('HELMCHART') else helm_chart_path
+    if for_preonboarding_checks:
+        # helm_chart_path = os.path.join(chart_export_path, 'connect-precheck-diagnoser')
+        helm_chart_path = os.path.join(chart_export_path, 'pre-onboarding-inspector')
+    else:
+        helm_chart_path = os.path.join(chart_export_path, 'azure-arc-k8sagents')
+
+    if for_preonboarding_checks:
+        chart_path = helm_chart_path
+    else:
+        chart_path = os.getenv('HELMCHART') if os.getenv('HELMCHART') else helm_chart_path
+
     return chart_path
 
 
-def pull_helm_chart(registry_path, kube_config, kube_context, helm_client_location):
+def pull_helm_chart(registry_path, kube_config, kube_context, helm_client_location, for_preonboarding_checks=False):
```

Comment:
use chart_name instead of this parameter, and you can set the default value of chart_name to AzureArcCharts. You can use single log message in this case by using the parameter name